ops/multidir_broadcast.go

```diff
@@ -92,14 +96,21 @@ func repeatTensorsForMutltidirBroadcast(A, B tensor.Tensor) (tensor.Tensor, tens
 // The given tensor is cloned such that the tensor is not modified in place.
 // Example: if we add 2 extra dimensions to shape (2, 3) we get shape (1, 1, 2, 3).
 func addExtraDimsToTensor(t tensor.Tensor, nExtraDims int) (tensor.Tensor, error) {
-	t = t.Clone().(tensor.Tensor)
+	t, ok := t.Clone().(tensor.Tensor)
+	if !ok {
+		return nil, ErrTypeAssert("tensor.Tensor", t.Clone())
```

Comment:
will this break? Because if the type assertion fails, t will be `nil` right? So then this might panic